text/0032-minimal-relay.md

```diff
@@ -0,0 +1,190 @@
+#### Staking
+
+Migrating the staking subsystem will likely be the most complex technical undertaking, as the
+Staking system cannot stop (the system MUST always have a validator set) nor run in parallel (the
+system MUST have only _one_ validator set) and the subsystem itself is made up of subsystems in the
+runtime and the node. For example, if offences are reported to the Staking parachain, validator
+nodes will need to submit their reports there.
+
+Handling balances also introduces complications. The same balance can be used for staking and
+governance. Ideally, all balances stay on Asset Hub, and only report "credits" to system chains like
+Staking and Governance. However, staking mutates balances by issuing new DOT on era changes and for
+rewards. Allowing DOT directly on the Staking parachain would simplify staking changes.
+
+Given the complexity, it would be pragmatic to include the Balances pallet in the Staking parachain
+in its first version. Any other systems that use overlapping locks, most notably governance, will
+need to recognise DOT held on both Asset Hub and the Staking parachain.
+
+There is more discussion about staking in a parachain in [Moving Staking off the Relay
+Chain](https://github.com/paritytech/polkadot-sdk/issues/491).
+
+#### Governance
+
+Migrating governance into a parachain will be less complicated than staking. Most of the primitives
+needed for the migration already exist. The Treasury supports spending assets on remote chains and
+collectives like the Polkadot Technical Fellowship already function in a parachain. That is, XCM
+already provides the ability to express system origins across chains.
+
+Therefore, actually moving the governance logic into a parachain will be simple. It can run in
+parallel with the Relay Chain's governance, which can be removed when the parachain has demonstrated
+sufficient functionality. It's possible that the Relay Chain maintain a Root-level emergency track
+for situations like [parachains
+halting](https://forum.polkadot.network/t/stalled-parachains-on-kusama-post-mortem/3998).
+
+The only complication arises from the fact that both Asset Hub and the Staking parachain will have
+DOT balances; therefore, the Governance chain will need to be able to credit users' voting power
+based on balances from both locations. This is not expected to be difficult to handle.
+
+## Drawbacks
+
+None at present.
+
+## Testing, Security, and Privacy
+
+Standard audit/review requirements apply. More powerful multi-chain integration test tools would be
+useful in developement.
+
+## Performance, Ergonomics, and Compatibility
+
+Describe the impact of the proposal on the exposed functionality of Polkadot.
+
+### Performance
+
+This is an optimization. The removal of public/user transactions on the Relay Chain ensures that its
+primary resources are allocated to system performance.
+
+### Ergonomics
+
+This proposal alters very little for coretime users (e.g. parachain developers). Application
```

Comment:
Parachains that interact with identity/staking/governance will need to coordinate with the migration to avoid outage.

Reply:
We'd ideally do XCMP or similar first because those messages being kept off the relay chain maybe more important than moving some tooling off.